### ISSUE TITLE
Mentioning a user in a message with a quote breaks formatting in the dmail

### ISSUE THREAD

**Issue by nottalulah** [danbooru/danbooru#5717]
[Example](https://testbooru.donmai.us/dmails/132?key=eyJfcmFpbHMiOnsiZGF0YSI6MTMyLCJwdXIiOiJkbWFpbF9saW5rIn19--8a6e4d7f46dc8900d8ecefa40f888c5265009c2f66b14bee6395b00821277537)

It looks like for some reason it strips out one of the opening [quote]s so it doesn't keep the main comment in a quote and has a unmatched closing element.

**Comment by nonamethanks** [comment]
The problem is that we use Rails' excerpt to cut off a message around the mention before sending it

https://github.com/danbooru/danbooru/blob/bd547ee9573357508649c09e6294dbab1903b389/app/logical/d_text.rb#L722-L728

So for:

```
[quote]
    Sturmgewehr44 said:

    Wait until you see that weirdo on Danbooru who kept begging for gun model implications.
[/quote]
@user Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
```

We're taking 

```
Wait until you see that weirdo on Danbooru who kept begging for gun model implications.
[/quote]
@user Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.

```

And then we surround the whole thing with [quote][/quote], which means that the first quote is matched to the middle quote, and then we have an extra [/quote], so obviously it breaks.

**Comment by ghost** [comment]
How about discarding the quoted text then?
```ruby
ActionController::Base.helpers.excerpt(strip_blocks("quote").to_s.gsub(/\r\n|\r|\n/, "\n"), needle, separator: "\n\n", radius: 1, omission: "")
```

**Comment by nonamethanks** [comment]
But then this: 

<img width="526" height="142" alt="Image" src="https://github.com/user-attachments/assets/6bb3f4a8-e3ee-44b1-bbf8-70f7aa48d7d9" />

turns into this: 

<img width="1352" height="513" alt="Image" src="https://github.com/user-attachments/assets/1c5a38fe-9cd8-4f75-9ff7-2ff6b04d81ba" />

**Comment by nonamethanks** [comment]
TBH I'm not even sure why we're even just sending an excerpt here. The dmail doesn't even mention that it's an excerpt, if you open the dmail and read it and then go to the link, if you're distracted, you might not even realize that it's a truncated message.

**Comment by ghost** [comment]
Yeah my initial thought was it should just not excerpt, but I assumed it was there for a reason, maybe size concerns.
The thought behind just stripping the quotes was it was simpler than trying to handle excerpting the quote, and you're missing the context regardless which requires you open the linked comment/forum to see what it means anyway. It also seems a bit semantically wrong, since excerpt is supposed to only fetch the paragraph with the mention in it, which I would not expect to include a quote (which itself would be a separate paragraph).
On the other hand, not excerpting could run into #5411, whereas if you have to click the link to view the rest you'd be more likely to reply there.

**Comment by nonamethanks** [comment]
We could just remove the reply button from danboorubot to prevent people from doing that. Users who wanted to take notes using it could still do so, but a cosmetic removal from received dmails would go a long way I think.

**Comment by nonamethanks** [comment]
Evazion pointed out in the discord that it might be a bad idea to do this because it would be more likely to trigger spam/adult filters when sending dmails. We've had problems with email delivery in the past so I'm hesitant in doing this. I guess an alternative would be to just make it clearer that the post being sent is an excerpt, by adding delimiters etc.
